common/src/test/java/org/jboss/da/common/version/VersionComparatorTest.java

```diff
@@ -121,6 +132,19 @@ public void testBaseVersion() {
         assertTrue(vc.compare("2.2.0.MR1", "2.2.0.CR1") < 0);
     }
 
+    @Test
+    public void testBaseVersionWithMultipleSuffixes() {
+        VersionComparator vc1 = new VersionComparator("3.4.2.Final", new VersionParser("temporary-redhat", "redhat"));
+
+        assertTrue(vc1.compare("3.4.2.Final", "3.4.2.Final-redhat-1") < 0);
```

Comment:
Sorry, not sure I understand, which version of the 2 "wins"? is it the first one or the second?  If it's the first I understand e.g. vc1.compare("3.4.2.Final-redhat-2", "3.4.2.Final-redhat-1") < 0 but not vc1.compare("3.4.2.Final", "3.4.2.Final-redhat-1") < 0

Reply:
You can read:
```
assertTrue(vc1.compare("3.4.2.Final", "3.4.2.Final-redhat-1") < 0);
```
as if you move the inequality sign between the strings:
```
"3.4.2.Final" < "3.4.2.Final-redhat-1"
```
i.e. "3.4.2.Final" is closer to the "base" version "3.4.2.Final" (which is true because they are equal :D)

```
"3.4.2.Final-redhat-2" < "3.4.2.Final-redhat-1"
```
"3.4.2.Final-redhat-2" wins here because if the versions are the same except for the suffix number, highest suffix number wins.

These comparsions to the base are for suggesting the best replacement for users. And in practice only compare versions with suffix.

Reply:
ook, I missed the "is closer to the "base" version" part. Understood now, thanks!